Rename runtimes/tr.source to runtimes/compiler

fjeremic opened the pull request:
Closes: #550
Signed-off-by: Filip Jeremic <fjeremic@ca.ibm.com>

fjeremic commented:
I have tested this commit using the OpenJ9 build instructions detailed in:
https://www.eclipse.org/openj9/oj9_build.html

I have also tested the change with an internal build, only building the JIT. I believe I will need to work with someone like @AdamBrousseau to sort out the dependencies on some of the infra tooling which may be affected by this change.

fjeremic commented:
Referencing comment https://github.com/eclipse/openj9/issues/550#issuecomment-344668184 it seems the travis build has hit the `jit.version` issue. @AdamBrousseau any pointers to how one would address this?

> jitVersionFileNotFound: compiler/jit.version

I will need to fix the copyright checks next.

jdekonin commented:
@fjeremic, you need to also change closed/OpenJ9.gmk references in https://github.com/ibmruntimes/openj9-openjdk-jdk9

fjeremic commented:
To all committers: I just parsed through every single file where I had an actual non-trivial change, i.e. excluding file renaming. All the copyrights have 2017 on the date line. I think the tool may be confusing the file renaming without any actual change within the file and flagging all source files as having incorrect copyright dates.

Do we consider directory renaming as a change which needs a copyright update? Intuitively I would think know but I'm ignorant on this subject.

fjeremic commented:
> @fjeremic, you need to also change closed/OpenJ9.gmk references in https://github.com/ibmruntimes/openj9-openjdk-jdk9

Excellent, this is what I was looking for. Any way to get a travis build with such a change on my personal fork? I understand I will also have to address internal repository "tr.source" to "compiler" renames as well.

jdekonin commented:
The travis build will fail because the repo isn't flexible enough to consume a 'depends' repo option.  But the jenkins pull request build can be performed that way.

For example: ```jenkins test sanity depends ibmruntimes/openj9-openjdk-jdk9#123```

jdekonin commented:
The openj9/.travis.yml will need updating as well.

AdamBrousseau commented:
re copyrights see #543 . Doesn't actually answer your question though

fjeremic commented:
> The openj9/.travis.yml will need updating as well.

Could you be more specific about this one? What in there needs to be modified?

AdamBrousseau commented:
I don't believe `.travis.yml` needs to be changed

fjeremic commented:
Jenkins test sanity depends ibmruntimes/openj9-openjdk-jdk9#86

fjeremic commented:
I split the changes into two commits. The first commit for the actual renaming in the various *.mk and *.xml files, etc., and a second commit for the actual rename of the directory. This is done to ease merge conflicts as every PR that gets merged in the JIT component will have a conflict with this PR.

fjeremic commented:
Jenkins test sanity depends ibmruntimes/openj9-openjdk-jdk9#86

fjeremic commented:
> re copyrights see #543 . Doesn't actually answer your question though

Thanks for this @AdamBrousseau. I did the same diff against my branch:

```
fjeremic@FILIP-IBM:~/git/openj9/runtime$ git diff -C --diff-filter=ACM --name-only github-upstream/master HEAD
buildspecs/aix_ppc-64.spec
buildspecs/aix_ppc-64_cmprssptrs.spec
buildspecs/aix_ppc-64_codecov.spec
buildspecs/aix_ppc.spec
buildspecs/j9.build-info
buildspecs/linux_390-64.spec
buildspecs/linux_390-64_cmprssptrs.spec
buildspecs/linux_390-64_cmprssptrs_codecov.spec
buildspecs/linux_390-64_codecov.spec
buildspecs/linux_390.spec
buildspecs/linux_arm.spec
buildspecs/linux_ppc-64.spec
buildspecs/linux_ppc-64_cmprssptrs.spec
buildspecs/linux_ppc-64_cmprssptrs_le.spec
buildspecs/linux_ppc-64_cmprssptrs_le_gcc.spec
buildspecs/linux_ppc-64_le.spec
buildspecs/linux_ppc-64_le_gcc.spec
buildspecs/linux_ppc.spec
buildspecs/linux_x86-64.spec
buildspecs/linux_x86-64_cmprssptrs.spec
buildspecs/linux_x86-64_cmprssptrs_cs.spec
buildspecs/linux_x86-64_cmprssptrs_panama.spec
buildspecs/linux_x86-64_codecov.spec
buildspecs/linux_x86.spec
buildspecs/linux_x86_codecov.spec
buildspecs/linux_ztpf_390-64.spec
buildspecs/linux_ztpf_390-64_cmprssptrs.spec
buildspecs/osx_x86-64.spec
buildspecs/win_x86-64.spec
buildspecs/win_x86-64_cmprssptrs.spec
buildspecs/win_x86.spec
buildspecs/zos_390-64.spec
buildspecs/zos_390-64_cmprssptrs.spec
buildspecs/zos_390.spec
runtime/buildtools.mk
runtime/cassume/module.xml
runtime/codert_vm/module.xml
runtime/ddr/module.xml
runtime/jit_vm/module.xml
runtime/jvmti/module.xml
runtime/makelib/targets.mk.ftl
runtime/makelib/targets.mk.linux.inc.ftl
runtime/module.xml
runtime/verbose/module.xml
runtime/windbg/module.xml
sourcetools/com.ibm.uma/com/ibm/j9/uma/Main.java
```

Then compared to the https://ci.eclipse.org/openj9/job/PullRequest-CopyrightCheck/1156/consoleFull console output and manually grepped each file. All the files in the above list have proper copyright dates.

DanHeidinga commented:
@0xdaryl mentioned potential problems with the symmetric naming approach in https://github.com/eclipse/openj9/issues/550#issuecomment-344697797 
> there might be subtle problems with a symmetric scheme that we would need to fix.
>
>I seem to recall there was a good reason why we didn't do it that way, but it escapes me right now. 

Have those issues been figured out / resolved?  I'd like to see a clear statement confirming that the designers of the "extensible classes" pattern believe this is the right approach.

@fjeremic Can you mark this as WIP until that occurs?

fjeremic commented:
> Have those issues been figured out / resolved? I'd like to see a clear statement confirming that the designers of the "extensible classes" pattern believe this is the right approach.

@DanHeidinga I think @0xdaryl was referring to his previous https://github.com/eclipse/openj9/issues/550#issuecomment-344695397 where he suggested the more symmetric approach is to fold the `trj9` directory as well so our paths look like:

```
runtimes/compiler/<stem path>
```

However this is now what my PR is trying to accomplish. We do not fold the `trj9` directory here, so all include paths that reference files in OpenJ9 have `trj9` prefix if we want the OpenJ9 version as opposed to the OMR one. Examples:

https://github.com/eclipse/openj9/blob/fcd9985beba1eb9c5b4be5171142e4d8efb47750/runtime/tr.source/trj9/codegen/J9TreeEvaluator.cpp#L36

https://github.com/eclipse/openj9/blob/master/runtime/tr.source/trj9/p/codegen/J9TreeEvaluator.cpp#L70

You will also note the keyword "tr.source" does not appear in any of the include paths directives in any of the OpenJ9 files, thus the name is irrelevant in this case. It is only the order of the include paths on the actual compiler directive (`g++ -I ...`) that matter.

@0xdaryl is my reasoning correct?

fjeremic commented:
Both sanity builds passed. Linking them here:
https://ci.eclipse.org/openj9/job/PullRequest-Sanity-linux_390-64_cmprssptrs/148/
https://ci.eclipse.org/openj9/job/PullRequest-Sanity-linux_ppc-64_cmprssptrs_le/151/

 because I made a change in https://github.com/ibmruntimes/openj9-openjdk-jdk9/pull/86 and would like to re-launch the builds.

Jenkins test sanity depends ibmruntimes/openj9-openjdk-jdk9#86

fjeremic commented:
All PRs are ready for merging. Last issue to address is the concerns noted above.

fjeremic commented:
https://github.com/ibmruntimes/openj9-openjdk-jdk9/pull/86 has been merged so in theory we should be able to pass the continuous builds now without a depends clause. Let's give it a shot:

Jenkins test sanity all

fjeremic commented:
Jenkins servers were down last week. Retrying again:

Jenkins test sanity

fjeremic commented:
Sanity builds all passed. Last concern to address is the one in https://github.com/eclipse/openj9/pull/589#issuecomment-344978858 to which I have replied in https://github.com/eclipse/openj9/pull/589#issuecomment-345005762. @0xdaryl any concerns regarding my answer? Did I interpret your post correctly?

Also could we get this assigned to one of the committers please?

0xdaryl commented:
I looked through the OpenJ9 code and its #include usage and from an extensible class ipath perspective this change is fine.

If the path name is subsequently changed (and I'm in support of that) then we need to be careful about referencing `compiler/` in the `#include` path name because the OMR project uses `compiler` as its project name and we would end up potentially including the wrong files.  Scanning through the code I think we'll be OK, but just something to be careful of.  That doesn't affect this PR though.

0xdaryl commented:
I am reviewing this PR, so please don't merge it yet.  It does need to be rebased though.

fjeremic commented:
> I looked through the OpenJ9 code and its #include usage and from an extensible class ipath perspective this change is fine.
> 
> If the path name is subsequently changed (and I'm in support of that) then we need to be careful about referencing compiler/ in the #include path name because the OMR project uses compiler as its project name and we would end up potentially including the wrong files. Scanning through the code I think we'll be OK, but just something to be careful of. That doesn't affect this PR though.

Thanks @0xdaryl. I have addressed the copyright change in https://github.com/eclipse/openj9/pull/589#issuecomment-344977150. I will rebase now to eliminate the conflicts.

0xdaryl commented on runtime/codert_vm/module.xml:
```diff
@@ -59,10 +59,10 @@
-			<include path="../tr.source/trj9/runtime" type="relativepath"/>
+			<include path="../compiler/runtime" type="relativepath"/>
+			<include path="../compiler/codegen" type="relativepath"/>
+			<include path="../compiler/base" type="relativepath"/>
```
The above 3 paths no longer exist in OpenJ9 and can be deleted.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/verbose/module.xml:
```diff
@@ -81,19 +81,19 @@
 			<include path="$(OMR_DIR)/gc/include" type="relativepath"/>
 			<include path="j9gcgluejava"/>
-			<include path="../tr.source/runtime" type="relativepath">
+			<include path="../compiler/runtime" type="relativepath">
```
This isn't a path that exists even in the current source.  I think it can be deleted.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/verbose/module.xml:
```diff
@@ -81,19 +81,19 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm/runtime" type="relativepath">
+			<include path="../compiler/ibm/runtime" type="relativepath">
```
This isn't a path that exists even in the current source.  I think it can be deleted.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/verbose/module.xml:
```diff
@@ -81,19 +81,19 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/codegen" type="relativepath">
+			<include path="../compiler/codegen" type="relativepath">
```
This isn't a path that exists even in the current source.  I think it can be deleted.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/verbose/module.xml:
```diff
@@ -81,19 +81,19 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm/codegen" type="relativepath">
+			<include path="../compiler/ibm/codegen" type="relativepath">
```
This isn't a path that exists even in the current source.  I think it can be deleted.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/verbose/module.xml:
```diff
@@ -102,22 +102,22 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/base" type="relativepath">
+			<include path="../compiler/base" type="relativepath">
```
This isn't a path that exists even in the current source.  I think it can be deleted.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/verbose/module.xml:
```diff
@@ -102,22 +102,22 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm" type="relativepath">
+			<include path="../compiler/ibm" type="relativepath">
```
This isn't a path that exists even in the current source.  I think it can be deleted.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/verbose/module.xml:
```diff
@@ -102,22 +102,22 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm/base" type="relativepath">
+			<include path="../compiler/ibm/base" type="relativepath">
```
This isn't a path that exists even in the current source.  I think it can be deleted.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/windbg/module.xml:
```diff
@@ -42,12 +42,12 @@
+			<include path="../compiler/codegen" type="relativepath"/>
+			<include path="../compiler/base" type="relativepath"/>
+			<include path="../compiler/codegen/sched" type="relativepath"/>
+			<include path="../compiler/optimizer" type="relativepath"/>
```
The above 4 paths no longer exists.  Delete.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on runtime/windbg/module.xml:
```diff
@@ -42,12 +42,12 @@
+			<include path="../compiler/optimizer" type="relativepath"/>
 			<include path="j9codert_vm"/>
-			<include path="../tr.source/runtime" type="relativepath"/>
+			<include path="../compiler/runtime" type="relativepath"/>
```
This path does not exist any longer.  Delete.

fjeremic replied:
Fixed in 626683a

0xdaryl commented on sourcetools/com.ibm.uma/com/ibm/j9/uma/Main.java:
```diff
@@ -224,9 +224,9 @@ public static void main(String[] args) {
+			File tempJitVersionFile = new File(rootDir + "/compiler/codegen/version.h");
 			if(tempJitVersionFile.exists() && !tempJitVersionFile.isDirectory()) {
-				jitVersionFile = rootDir + "/tr.source/codegen/version.h";
+				jitVersionFile = rootDir + "/compiler/codegen/version.h";
```
Neither of these paths exist any more.  If it still needs to run perhaps the initial path should be `/compiler/trj9/build/version.h`.

fjeremic replied:
Fixed in 626683a

fjeremic commented on runtime/cassume/module.xml:
```diff
@@ -58,15 +58,15 @@
-			<include path="../tr.source/ibm/runtime" type="relativepath"/>
+			<include path="../compiler/frontend/j9" type="relativepath"/>
+			<include path="../compiler/runtime" type="relativepath"/>
+			<include path="../compiler/ibm/runtime" type="relativepath"/>
```
Fixed in 626683a

fjeremic commented on runtime/cassume/module.xml:
```diff
@@ -58,15 +58,15 @@
+			<include path="../compiler/codegen" type="relativepath"/>
+			<include path="../compiler/ibm/codegen" type="relativepath"/>
+			<include path="../compiler/base" type="relativepath"/>
+			<include path="../compiler/ibm/base" type="relativepath"/>
```
Fixed in 626683a

fjeremic commented:
> I made a number of comments on paths that no longer exist. While I think they should be fixed, I'm OK if they don't get fixed in this PR but a new PR is created to clean them up right after.
> 
> The PR does need to be rebased to resolve merge conflicts.

No better time than now. Rebased and addressed all concerns. Given the nature and scope of the change I wouldn't feel comfortable merging without another full Jenkins run and an internal build.

Jenkins test sanity

0xdaryl commented on runtime/ddr/module.xml:
```diff
@@ -81,28 +81,28 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/base" type="relativepath">
+			<include path="../compiler/base" type="relativepath">
```
This path doesn't exist in OpenJ9.

fjeremic replied:
Fixed in 9ba3ffe

0xdaryl commented on runtime/ddr/module.xml:
```diff
@@ -81,28 +81,28 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm/base" type="relativepath">
+			<include path="../compiler/ibm/base" type="relativepath">
```
This path doesn't exist in OpenJ9.

fjeremic replied:
Fixed in 9ba3ffe

0xdaryl commented on runtime/ddr/module.xml:
```diff
@@ -81,28 +81,28 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/env" type="relativepath">
+			<include path="../compiler/env" type="relativepath">
```
This path doesn't exist in OpenJ9.

fjeremic replied:
Fixed in 9ba3ffe

0xdaryl commented on runtime/ddr/module.xml:
```diff
@@ -81,28 +81,28 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm/env" type="relativepath">
+			<include path="../compiler/ibm/env" type="relativepath">
```
This path doesn't exist in OpenJ9.

0xdaryl commented on runtime/jit_vm/module.xml:
```diff
@@ -60,12 +60,12 @@
+			<include path="../compiler/base" type="relativepath"/>
+			<include path="../compiler/optimizer" type="relativepath"/>
+			<include path="../compiler/frontend/j9" type="relativepath"/>
+			<include path="../compiler/runtime" type="relativepath"/>
```
These paths don't exist in OpenJ9.  Delete.

0xdaryl commented on runtime/jvmti/module.xml:
```diff
@@ -71,25 +71,25 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm/runtime" type="relativepath">
+			<include path="../compiler/ibm/runtime" type="relativepath">
```
This path doesn't exist in OpenJ9.

0xdaryl commented on runtime/jvmti/module.xml:
```diff
@@ -71,25 +71,25 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/codegen" type="relativepath">
+			<include path="../compiler/codegen" type="relativepath">
```
This path doesn't exist in OpenJ9.

0xdaryl commented on runtime/jvmti/module.xml:
```diff
@@ -71,25 +71,25 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm/codegen" type="relativepath">
+			<include path="../compiler/ibm/codegen" type="relativepath">
```
This path doesn't exist in OpenJ9.

0xdaryl commented on runtime/jvmti/module.xml:
```diff
@@ -71,25 +71,25 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>			
-			<include path="../tr.source/base" type="relativepath">
+			<include path="../compiler/base" type="relativepath">
```
This path doesn't exist in OpenJ9.

fjeremic replied:
Fixed in 9ba3ffe

0xdaryl commented on runtime/jvmti/module.xml:
```diff
@@ -71,25 +71,25 @@
 				<include-if condition="spec.flags.J9VM_INTERP_NATIVE_SUPPORT"/>
 			</include>
-			<include path="../tr.source/ibm/base" type="relativepath">
+			<include path="../compiler/ibm/base" type="relativepath">
```
This path doesn't exist in OpenJ9.

fjeremic replied:
Fixed in 9ba3ffe

0xdaryl commented on runtime/makelib/targets.mk.linux.inc.ftl:
```diff
@@ -418,7 +418,7 @@ endif
 <#if uma.spec.processor.arm>
 $(patsubst %.s,%.o,$(filter %.s,$(UMA_FILES_TO_PREPROCESS))): %$(UMA_DOT_O): %.s
-	sed -f $(UMA_PATH_TO_ROOT)tr.source/codegen/makefiles/armasm2gas.sed $*.s > $*.S
+	sed -f $(UMA_PATH_TO_ROOT)compiler/codegen/makefiles/armasm2gas.sed $*.s > $*.S
```
The path should be `compiler/trj9/build/scripts/armasm2gas.sed`

fjeremic replied:
Fixed in 9ba3ffe

fjeremic commented:
All concerns addressed again. Relaunching builds:

Jenkins test sanity
